### PR TITLE
{KeyVault} Expose resource name directly

### PR DESCRIPTION
**Pull request by bim-msft**
This is for supporting: #11538
Support `key`, `secret`, `certificate` in this PR.

Other services like vm or storage account are exposing their resource name directly, but keyvault is not.

For example:
```
(env3) C:\repos\azure-cli>az vm show -n bim-vm -g bim-rg
{
   ...
  "id": "/subscriptions/0b1f6471-1bf0-4dda-aec3-cb9272f09590/resourceGroups/bim-rg/providers/Microsoft.Compute/virtualMachines/bim-vm",
  "identity": null,
  "instanceView": null,
  "licenseType": null,
  "location": "japaneast",
  "name": "bim-vm",

...
```

```
(env3) C:\repos\azure-cli>az storage account show -n bimstorageacc -g bim-rg
{
  ...
  "id": "/subscriptions/0b1f6471-1bf0-4dda-aec3-cb9272f09590/resourceGroups/bim-rg/providers/Microsoft.Storage/storageAccounts/bimstorageacc",
  "identity": null,
  "isHnsEnabled": null,
  "kind": "StorageV2",
  "largeFileSharesState": null,
  "lastGeoFailoverTime": null,
  "location": "westcentralus",
  "name": "bimstorageacc",
...
```

---

This checklist is used to make sure that common guidelines for a pull request are followed.

- [x] The PR title and description has followed the guideline in [Submitting Pull Requests](https://github.com/Azure/azure-cli/tree/dev/doc/authoring_command_modules#submitting-pull-requests).

- [x] I adhere to the [Command Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/command_guidelines.md).

### CODE REVIEW

**Comment by jiasli** [comment]
Better to post some examples - Before and After.

**Comment by bim-msft** [comment]
> Better to post some examples - Before and After.

@jiasli 
Command: `az keyvault secret list --vault-name bim-kv5`
Before:
```
[
  {
    "attributes": {
      "created": "2020-03-10T10:47:18+00:00",
      "enabled": false,
      "expires": null,
      "notBefore": null,
      "recoveryLevel": "CustomizedRecoverable",
      "updated": "2020-03-10T10:48:20+00:00"
    },
    "contentType": null,
    "id": "https://bim-kv5.vault.azure.net/secrets/test-s4",
    "managed": null,
    "tags": {
      "file-encoding": "utf-8"
    }
  }
]
```

After (added "name" field):
```
[
  {
    "attributes": {
      "created": "2020-03-10T10:47:18+00:00",
      "enabled": false,
      "expires": null,
      "notBefore": null,
      "recoveryLevel": "CustomizedRecoverable",
      "updated": "2020-03-10T10:48:20+00:00"
    },
    "contentType": null,
    "id": "https://bim-kv5.vault.azure.net/secrets/test-s4",
    "managed": null,
    "name": "test-s4",
    "tags": {
      "file-encoding": "utf-8"
    }
  }
]
```

Also supported on command groups `key`, `certificate`, and the commands `show`, `delete`, .etc.

**Comment by bim-msft** [comment]
Wrong merge operation, sorry for disturbance. Please ignore the review request.

File: src/azure-cli/azure/cli/command_modules/keyvault/_validators.py [hunk 1]
```diff
@@ -374,3 +375,22 @@ def _validate(ns):
...
+        resource_name = items[4] if len(items) > 4 else None
+
+    setattr(output, 'name', resource_name)
+    return output
+
+
+def extract_subresource_name(id_parameter='id'):
+    def _extract_subresource_name(output):
+        if isinstance(output, Iterable):
+            return [_extract_subresource_name_from_single_output(_, id_parameter) for _ in output]
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/keyvault/_validators.py]: Underscore `_` usually means the value will be dropped. If it is used in another function call, we'd better give it some meaningful name.

Comment (bim-msft) [reply src/azure-cli/azure/cli/command_modules/keyvault/_validators.py]: Really good point, renamed it to `item`.

File: src/azure-cli/azure/cli/command_modules/keyvault/_transformers.py [hunk 2]
```diff
@@ -3,6 +3,35 @@
...
 def filter_out_managed_resources(output):
     return [_ for _ in output if not getattr(_, 'managed')] if output else output
+
+
+def _extract_subresource_name_from_single_output(output, id_parameter):
+    if not getattr(output, id_parameter):
+        resource_name = None
+    else:
+        items = getattr(output, id_parameter).split('/')
+        resource_name = items[4] if len(items) > 4 else None
```
Comment (fengzhou-msft) [inline src/azure-cli/azure/cli/command_modules/keyvault/_transformers.py]: Can we use `parse_resource_id` here?

Comment (bim-msft) [reply src/azure-cli/azure/cli/command_modules/keyvault/_transformers.py]: > `parse_resource_id`

@fengzhou-msft Good idea! I will use `parse_resource_id`.

**Pull request merged**